gcloud-java-dns/src/main/java/com/google/gcloud/dns/Zone.java

```diff
@@ -68,166 +66,81 @@ public static Zone get(Dns dnsService, String zoneName, Dns.ZoneOption... option
   }
 
   /**
-   * Constructs a {@code Zone} object that contains information received from the Google Cloud DNS
-   * service for the provided {@code zoneId}.
-   *
-   * @param zoneId ID of the zone to be searched for
-   * @param options optional restriction on what fields should be returned by the service
-   * @return zone object containing zone's information or {@code null} if not not found
-   * @throws DnsException upon failure
-   */
-  public static Zone get(Dns dnsService, BigInteger zoneId, Dns.ZoneOption... options) {
-    checkNotNull(zoneId);
-    checkNotNull(dnsService);
-    ZoneInfo zoneInfo = dnsService.getZone(zoneId, options);
-    return zoneInfo == null ? null : new Zone(dnsService, zoneInfo);
-  }
-
-  /**
-   * Retrieves the latest information about the zone. The method first attempts to retrieve the zone
-   * by ID and if not set or zone is not found, it searches by name.
+   * Retrieves the latest information about the zone. The method retrieves the zone by name which
+   * must always be initialized.
    *
    * @param options optional restriction on what fields should be fetched
    * @return zone object containing updated information or {@code null} if not not found
    * @throws DnsException upon failure
-   * @throws NullPointerException if both zone ID and name are not initialized
    */
   public Zone reload(Dns.ZoneOption... options) {
-    checkNameOrIdNotNull();
-    Zone zone = null;
-    if (zoneInfo.id() != null) {
-      zone = Zone.get(dns, zoneInfo.id(), options);
-    }
-    if (zone == null && zoneInfo.name() != null) {
-      // zone was not found by id or id is not set at all
-      zone = Zone.get(dns, zoneInfo.name(), options);
-    }
-    return zone;
+    return Zone.get(dns, zoneInfo.name(), options);
   }
 
   /**
-   * Deletes the zone. The method first attempts to delete the zone by ID. If the zone is not found
-   * or id is not set, it attempts to delete by name.
+   * Deletes the zone. The method first deletes the zone by name which must always be initialized.
```

Comment:
No need for "first". Also, name should always be available (checked by`ZoneInfo`).